framgia/sph-ojt-blog
# Add pagination and seed with dummy data using django_seed

**Issue Link**
Asana: https://app.asana.com/0/1202718495875213/1202725278795544/f

Definition of Done:
- Can make requests to the API as to what page number and number of blogs to display.
- Example: http://localhost:8000/api/posts?page=1&count=3
-The above request fetches: page 1 and displays 3 blogs

Steps to test functionality:

- I used the django-seed library to automatically seed dummy data to the models
- In your terminal, activate the virtual environment in the backend
- Then run: 
```
pip install django-seed
pip install psycopg2
```

- After the dependencies are installed, run:
`
python manage.py seed post --number=30 # 30 refers to the number of dummy data you want to add to the models
`
- Run the server: python manage.py runserver
- Go to postman or your browser then type: 
http://localhost:8000/api/posts?page=1&count=3 # 1 refers to page number and 5 refers to the number of blogs you want to display
- It should look something like this

![image](https://user-images.githubusercontent.com/108708114/184018766-22cede25-69ca-4e9a-b017-f5a6930e2f85.png)

Notes:
- django-seed could not build many-to-many relationships, so you need to do them manually if you want. However, this shouldn't affect the implementation.
